### PR TITLE
Feature/multi governed users

### PR DESCRIPTION
**Pull request by aleksey-kuznetsov**
This is the first portion of changes to extend current multi-governed user functionality

### CODE REVIEW

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (yufengwng) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Where can we find the backend changes for these new services?

**Comment by fat23cat** [comment]
Looks good! But I'd suggest don't merge until backend changes approved(to be sure that API won't be changed)

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (aleksey-kuznetsov) [reply ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Here: https://github.com/broadinstitute/ddp-study-server/pull/463
As @fat23cat suggested I'd wait for backend changes to be approved first.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts
Comment (yufengwng) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: From PR https://github.com/broadinstitute/ddp-study-server/pull/463, it looks like this API doesn't accept `alias` in the request payload. Even if it did, maybe setting it to the current participant's guid is not best way? The alias is meant to be kind of a "name" for the new governed participant so setting it to current participant is a bit odd.

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/activityServiceAgent.service.ts
Comment (yufengwng) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/activityServiceAgent.service.ts]: As I mentioned in the other PR's [comment](https://github.com/broadinstitute/ddp-study-server/pull/463#discussion_r513585969), how about we make the endpoint `/activities/${activityGuid}/answers/${questionStableId}`?

Comment (aleksey-kuznetsov) [reply ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/activityServiceAgent.service.ts]: Alright, we decided to remove this endpoint and therefore this change is not needed.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts
Comment (aleksey-kuznetsov) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: You are correct. This part of code was not written by me so I decided to change it so it behaves the same way it did before our changes. Anyway, I'll remove the part with alias.

Comment (aleksey-kuznetsov) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: We decided to keep this part, @YuriyPavlovBI will handle alias field if it is provided.

Comment (aleksey-kuznetsov) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: To clarify, we are not using this exact component in our AT study. These changes were made only to ensure that this component behaves same way as it did before.

Comment (yufengwng) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: I don't think it should pass in the current participant guid. That doesn't make sense. Let's keep it same for now and I'll add ticket to fix this separately.

**Review by yufengwng — COMMENTED** [review]
Looks good as far as I can tell. I'll let Pavel and Marco have the final say here.

**Comment by aleksey-kuznetsov** [comment]
@fat23cat @MocanaAtBroad please give us your feedback :)

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (MocanaAtBroad) [reply ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Looks like this PR got merged, but this call does not match the API.
Looked at the Java code and the generated REST API Docs. I can confirm that the docs match what the server is expecting:
https://pepper-api-spec-dot-broad-ddp-dev.uc.r.appspot.com/#operation/createGovernedStudyParticipant
There is no `alias` attribute expected in POST payload.

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Could we create or reuse an interface or class that that defines the expected payload to be sent?

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Suggest not returning an `HttpResponse`. Perhaps result could be `map`ed to a domain object, counter or something else that does include HTTP specific classes?

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/workflowServiceAgent.service.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/workflowServiceAgent.service.ts]: Does not appear that this is used anywhere. This is OK right now?

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/workflowServiceAgent.service.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/workflowServiceAgent.service.ts:38]: Same as above.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: Does not appear that `session` is being used. Please feel free to remove.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts:81]: github not letting me comment on it, but line 68:
can be changed to:
```typescript
mergeMap(x => this.serviceAgent.getList()),
```
Should fix deprecated warning I am seeing.
(I know this is not one of your changes, but since you are in there already...)

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Aware this was already here/not one of your changes, but does this work?
This is being called from `manageParticipansComponent. That URL path does not include the `/study`.
I assume this is the route we are trying to reach?
https://pepper-api-spec-dot-broad-ddp-dev.uc.r.appspot.com/#operation/getGovernedStudyParticipants

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts]: `switchMap` here is probably not a good idea. Leave open possibility that you could have created a user in the server but locally one of those created users would be lost.
Safer to use `concatMap` instead.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/manageParticipants.component.ts:100]: Consider adding a `take(1)` to pipeline after `tap()` to terminate subscription each time `add()` is executed.,

File: ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts
Comment (MocanaAtBroad) [reply ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/governedParticipantsServiceAgent.service.ts]: Still not seeing how `getList()` could work with URL path of just `/participants`

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/participantProfile.component.ts
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/participantProfile.component.ts]: Can we always expect that userProfile property will be set? (I don't know the answer).
If not sure or don't know then try to be defend for undefined or null condition here.

**Review by MocanaAtBroad — COMMENTED** [review]
Just one small comment. Otherwise looking good.

**Review by fat23cat — DISMISSED** [review]

**Review by fat23cat — APPROVED** [review]

**Pull request merged**
